src/apps/answers/service.py

```diff
@@ -366,6 +366,18 @@ async def validate_multiinformant_assessment(
             activity, flow = await asyncio.gather(activity_future, flow_future)
             if not activity and not flow:
                 raise MultiinformantAssessmentInvalidActivityOrFlow()
+            
+        is_admin = await AppletAccessCRUD(self.session).has_any_roles_for_applet(
+            applet_id,
+            respondent_subject.user_id,
+            [Role.OWNER],
+        )
+
+        if not is_admin:
+            if not target_subject or not source_subject:
+                raise MultiinformantAssessmentNoAccessApplet()
```

Comment:
Should there be a message included in this exception for helpful context? (`Missing target subject or source subject`)

Reply:
yes great point i have added the message!
and thank you for sharing the document with me paul !